{"priority": "trivial", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/hoffmanlab/segway.json"}, "html": {"href": "#!/hoffmanlab/segway"}, "avatar": {"href": "data/bytebucket.org/ravatar/{68bf38bf-f0c1-4c9c-909e-0ee4b2c26257}ts=python"}}, "type": "repository", "name": "segway", "full_name": "hoffmanlab/segway", "uuid": "{68bf38bf-f0c1-4c9c-909e-0ee4b2c26257}"}, "links": {"attachments": {"href": "data/repositories/hoffmanlab/segway/issues/17/attachments_page=1.json"}, "self": {"href": "data/repositories/hoffmanlab/segway/issues/17.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/hoffmanlab/segway/issues/17/watch"}, "comments": {"href": "data/repositories/hoffmanlab/segway/issues/17/comments_page=1.json"}, "html": {"href": "#!/hoffmanlab/segway/issues/17/unclear-error-message-when-using-space"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/hoffmanlab/segway/issues/17/vote"}}, "reporter": {"display_name": "ajshaw", "uuid": "{547650d6-1225-4731-b1da-ba9913babfbf}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B547650d6-1225-4731-b1da-ba9913babfbf%7D"}, "html": {"href": "https://bitbucket.org/%7B547650d6-1225-4731-b1da-ba9913babfbf%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/1f9b227ab720d7a41f4f0d49570db03fd=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsA-2.png"}}, "nickname": "ajshaw", "type": "user", "account_id": "5b22d060ba383e0814004a64"}, "title": "Unclear error message when using space-delimited segment table", "component": null, "votes": 0, "watches": 2, "content": {"raw": "If the user specifies a segment table with --seg-table=*FILE*, and the file separates by spaces rather than tabs (text editors sometimes replace tabs with spaces automatically depending on their configurations), segway fails on an assertion error which does not make clear what the problem is. Depending on whether this is common, it may help users to suggest that this may be the problem.\r\n\r\n\r\n```\r\nTraceback (most recent call last):\r\n  File \"/mnt/work1/users/home2/ashaw/bin/segway\", line 9, in <module>\r\n    load_entry_point('segway==1.2.0.dev-rel889-hg992cc54de504', 'console_scripts', 'segway')()\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 2537, in main\r\n    return runner()\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 2359, in __call__\r\n    self.run(*args, **kwargs)\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 2320, in run\r\n    self.save_observations_params()\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 1460, in save_observations_params\r\n    self.save_include()\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 1337, in save_include\r\n    IncludeSaver(self)(self.gmtk_include_filename, aux_dirpath,\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py\", line 136, in __init__\r\n    copy_attrs(runner, self, self.copy_attrs)\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py\", line 128, in copy_attrs\r\n    setattr(dst, attr, getattr(src, attr))\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py\", line 465, in __get__\r\n    res = self.fget(instance or owner)\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 1064, in card_seg_countdown\r\n    return self.seg_countdowns_initial.max() + 1\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py\", line 465, in __get__\r\n    res = self.fget(instance or owner)\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 1038, in seg_countdowns_initial\r\n    table = self.seg_table\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py\", line 465, in __get__\r\n    res = self.fget(instance or owner)\r\n  File \"/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py\", line 808, in seg_table\r\n    label = row[\"label\"]\r\nKeyError: 'label'\r\n\r\n\r\n```\r\n", "markup": "markdown", "html": "<p>If the user specifies a segment table with --seg-table=<em>FILE</em>, and the file separates by spaces rather than tabs (text editors sometimes replace tabs with spaces automatically depending on their configurations), segway fails on an assertion error which does not make clear what the problem is. Depending on whether this is common, it may help users to suggest that this may be the problem.</p>\n<div class=\"codehilite\"><pre><span></span>Traceback (most recent call last):\n  File &quot;/mnt/work1/users/home2/ashaw/bin/segway&quot;, line 9, in &lt;module&gt;\n    load_entry_point(&#39;segway==1.2.0.dev-rel889-hg992cc54de504&#39;, &#39;console_scripts&#39;, &#39;segway&#39;)()\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 2537, in main\n    return runner()\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 2359, in __call__\n    self.run(*args, **kwargs)\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 2320, in run\n    self.save_observations_params()\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 1460, in save_observations_params\n    self.save_include()\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 1337, in save_include\n    IncludeSaver(self)(self.gmtk_include_filename, aux_dirpath,\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py&quot;, line 136, in __init__\n    copy_attrs(runner, self, self.copy_attrs)\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py&quot;, line 128, in copy_attrs\n    setattr(dst, attr, getattr(src, attr))\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py&quot;, line 465, in __get__\n    res = self.fget(instance or owner)\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 1064, in card_seg_countdown\n    return self.seg_countdowns_initial.max() + 1\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py&quot;, line 465, in __get__\n    res = self.fget(instance or owner)\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 1038, in seg_countdowns_initial\n    table = self.seg_table\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/_util.py&quot;, line 465, in __get__\n    res = self.fget(instance or owner)\n  File &quot;/mnt/work1/users/home2/ashaw/bin/lib/python2.7/site-packages/segway-1.2.0.dev_rel889_hg992cc54de504-py2.7.egg/segway/run.py&quot;, line 808, in seg_table\n    label = row[&quot;label&quot;]\nKeyError: &#39;label&#39;\n</pre></div>", "type": "rendered"}, "assignee": null, "state": "new", "version": null, "edited_on": null, "created_on": "2014-07-28T08:07:08.893479+00:00", "milestone": null, "updated_on": "2014-10-16T14:53:44.419024+00:00", "type": "issue", "id": 17}